### PR TITLE
Application pdf updated

### PR DESCRIPTION
**Pull request by steffen12**
Making changes with the fields for applications in ApplicationForm

### CODE REVIEW

**Comment by loafyyy** [comment]
@steffen12 Can you give more detailed descriptions in your PRs or at least add some screenshots? Especially because you added a lot of code so it's harder to review. Thanks

File: src/components/AccountSecurity/IssueReport.tsx [hunk 1]
```diff
@@ -157,9 +157,12 @@ class IssueReport extends Component<Props, State, {}> {
   handleSubmitWithRecaptcha = async (event: any) => {
     event.preventDefault();
     if (recaptchaRef !== null && recaptchaRef.current !== null) {
+      // @ts-ignore
```
Comment (loafyyy) [inline src/components/AccountSecurity/IssueReport.tsx:160]: What's this and why is it commented out?

File: src/components/AccountSecurity/LoginPage.tsx [hunk 2]
```diff
@@ -365,7 +365,7 @@ class LoginPage extends Component<Props, State> {
                 </div>
                 <div className="row">
                   <div className="col-10 pl-0">
-                    <Link to="/find-organization">
+                    <Link to="/find-organizations">
```
Comment (loafyyy) [inline src/components/AccountSecurity/LoginPage.tsx:368]: I fixed this bug already -- please merge in the latest master

File: src/components/AccountSecurity/MyOrganization.tsx [hunk 3]
```diff
@@ -223,6 +223,7 @@ class MyOrganization extends Component<Props, State> {
             <option defaultValue="" disabled hidden aria-labelledby="role" />
             <option value="Admin">Admin</option>
             <option value="Worker">Worker</option>
+            <option value="Client">Client</option>
```
Comment (loafyyy) [inline src/components/AccountSecurity/MyOrganization.tsx:226]: Is this related to the Application PDF component? Usually, your PR should just focus on one page/topic to make your changes relevant to each other.

File: src/components/PDF/ApplicationForm.tsx [hunk 4]
```diff
@@ -51,7 +52,8 @@ class ApplicationForm extends Component<Props, State> {
       formQuestions: undefined,
       formAnswers: {},
       pdfApplication: undefined,
-      buttonState: '',
```
Comment (loafyyy) [inline src/components/PDF/ApplicationForm.tsx:54]: Why was buttonState removed?

File: src/components/PDF/ApplicationForm.tsx [hunk 5]
```diff
@@ -267,17 +297,27 @@ class ApplicationForm extends Component<Props, State> {
...
           <div className="container border px-5 col-lg-10 col-md-10 col-sm-12">
             <form onSubmit={this.onSubmitFormQuestions}>
               {formQuestions.map(
                 (entry) => (
-                  <div className="my-5" key={uuid()}>
+                  <div className="my-5" key={entry.fieldID}>
                     {
                       (() => {
+                        if (entry.fieldType === 'ReadOnlyField') {
+                          // TODO: Make it work
```
Comment (loafyyy) [inline src/components/PDF/ApplicationForm.tsx:312]: Why is this all commented out? Can you please remove commented out code that's not needed, otherwise, it makes people confused.

**Review by loafyyy — COMMENTED** [review]
I gave you some questions that I would like answered. Maybe we can call sometime so I can better understand the Applications page.

**Comment by steffen12** [comment]
> I gave you some questions that I would like answered. Maybe we can call sometime so I can better understand the Applications page.

Would be honored to call with loafffy

File: src/components/AccountSecurity/IssueReport.tsx (hunk 1 above)
Comment (steffen12) [reply src/components/AccountSecurity/IssueReport.tsx:160]: Without it it gives me a typescript error

File: src/components/AccountSecurity/LoginPage.tsx (hunk 2 above)
Comment (steffen12) [reply src/components/AccountSecurity/LoginPage.tsx]: ok

File: src/components/AccountSecurity/MyOrganization.tsx (hunk 3 above)
Comment (steffen12) [reply src/components/AccountSecurity/MyOrganization.tsx:226]: No sorry, it is a small change to allow for clients to be enrolled

File: src/components/PDF/ApplicationForm.tsx (hunk 4 above)
Comment (steffen12) [reply src/components/PDF/ApplicationForm.tsx:54]: Well, it is unused I believe because now we are transitioning to pagination on the questions, and so the button does not swirl (I got rid of the swirling button). I think we need to redo it now with pagination (Vanessa covered this)

File: src/components/PDF/ApplicationForm.tsx (hunk 5 above)
Comment (steffen12) [reply src/components/PDF/ApplicationForm.tsx]: It was there for the future, but I fixed it already

Comment (steffen12) [reply src/components/PDF/ApplicationForm.tsx]: Could we do this in the future, including leaving comments with code for us to work on in the future, and if not then what should we do instead?

**Comment by steffen12** [comment]
I added automatic form filling with pre-existing database values and also made it so that forms show a description and title now. Please let me know what you think - we need to merge soon so that other teams can use it. Thanks :)

File: src/components/SignUp/OrganizationInformation.tsx [hunk 6]
```diff
@@ -312,7 +312,7 @@ class OrganizationInformation extends Component<Props, State, {}> {
                 <label htmlFor="orgWebsite" className="col-sm-3 col-form-label text-sm-right">Organization website</label>
                 <div className="col-sm-9">
                   <input
-                    type="url"
+                    type="text"
```
Comment (crchong1) [inline src/components/SignUp/OrganizationInformation.tsx:315]: why is this a text field and not a url field anymore?

**Review by crchong1 — CHANGES_REQUESTED** [review]

File: src/components/PDF/ApplicationForm.tsx [hunk 7]
```diff
@@ -19,14 +19,14 @@ interface Props {
 }
 
 interface State {
-  formQuestions: any[] | undefined,
+  fields: any[] | undefined,
```
Comment (crchong1) [inline src/components/PDF/ApplicationForm.tsx:22]: In future PR please annotate the fields so we remove any[]s. This is bad practice because any[] circumvents strong typing which is the whole reason we are using typescript

File: src/components/PDF/ApplicationForm.tsx [hunk 8]
```diff
@@ -48,21 +48,22 @@ class ApplicationForm extends Component<Props, State> {
...
       numPages: 0,
-      startDate: new Date(),
       formError: false,
     };
     this.handleChangeFormValueTextField = this.handleChangeFormValueTextField.bind(this);
     this.handleChangeFormValueRadioButton = this.handleChangeFormValueRadioButton.bind(this);
     this.handleChangeFormValueCheckBox = this.handleChangeFormValueCheckBox.bind(this);
     this.handleChangeFormValueListBox = this.handleChangeFormValueListBox.bind(this);
-    this.onSubmitFormQuestions = this.onSubmitFormQuestions.bind(this);
+    this.handleChangeFormValueDateField = this.handleChangeFormValueDateField.bind(this);
```
Comment (loafyyy) [inline src/components/PDF/ApplicationForm.tsx:65]: Don't need to bind if using arrow functions

File: src/components/PDF/ApplicationForm.tsx [hunk 9]
```diff
@@ -48,21 +48,22 @@ class ApplicationForm extends Component<Props, State> {
...
-      buttonState: '',
+      title: '',
+      description: '',
       submitSuccessful: false,
       currentPage: 0,
       numPages: 0,
-      startDate: new Date(),
       formError: false,
     };
     this.handleChangeFormValueTextField = this.handleChangeFormValueTextField.bind(this);
```
Comment (crchong1) [inline src/components/PDF/ApplicationForm.tsx:61]: remove bind statements and transform them to arrow functions

File: src/components/PDF/ApplicationForm.tsx [hunk 10]
```diff
@@ -290,37 +333,57 @@ class ApplicationForm extends Component<Props, State> {
                                 id={entry.fieldName}
```
Comment (crchong1) [inline src/components/PDF/ApplicationForm.tsx:333]: fieldId

File: src/components/PDF/ApplicationForm.tsx [hunk 11]
```diff
@@ -290,37 +333,57 @@ class ApplicationForm extends Component<Props, State> {
...
+                        }
+                        if (entry.fieldType === 'MultilineTextField') {
+                          return (
+                            <div className="mt-2 mb-2">
+                              <label htmlFor={entry.fieldName} className="w-100 font-weight-bold">
+                                {entry.fieldQuestion}
+                              </label>
+                              <textarea
+                                className="form-control form-purple mt-1"
+                                id={entry.fieldName}
```
Comment (crchong1) [inline src/components/PDF/ApplicationForm.tsx:352]: fieldId

File: src/components/PDF/ApplicationForm.tsx [hunk 12]
```diff
@@ -205,12 +237,13 @@ class ApplicationForm extends Component<Props, State> {
...
+      title,
+      description,
+      fields,
       formAnswers,
       submitSuccessful,
       currentPage,
       numPages,
-      startDate,
       formError,
     } = this.state;
```
Comment (crchong1) [inline src/components/PDF/ApplicationForm.tsx:249]: alert.show('Successfully submitted ...', { type: 'success' }) <- something like this

File: src/components/SignUp/OrganizationInformation.tsx (hunk 6 above)
Comment (crchong1) [reply src/components/SignUp/OrganizationInformation.tsx:315]: dont change it to text. its already fine as is

**Review by crchong1 — CHANGES_REQUESTED** [review]

File: src/components/SignUp/OrganizationInformation.tsx (hunk 6 above)
Comment (steffen12) [reply src/components/SignUp/OrganizationInformation.tsx:315]: Fixed

File: src/components/PDF/ApplicationForm.tsx (hunk 8 above)
Comment (steffen12) [reply src/components/PDF/ApplicationForm.tsx:65]: Done.

File: src/components/PDF/ApplicationForm.tsx (hunk 7 above)
Comment (steffen12) [reply src/components/PDF/ApplicationForm.tsx:22]: Done

File: src/components/PDF/ApplicationForm.tsx (hunk 9 above)
Comment (steffen12) [reply src/components/PDF/ApplicationForm.tsx:61]: Done

File: src/components/PDF/ApplicationForm.tsx (hunk 10 above)
Comment (steffen12) [reply src/components/PDF/ApplicationForm.tsx:333]: Well, actually it works with fieldName because fieldID is frontend ID (just to satisfy react complaining about keys), while fieldName is the backend ID that is used to identify the fields on the server

File: src/components/PDF/ApplicationForm.tsx (hunk 11 above)
Comment (steffen12) [reply src/components/PDF/ApplicationForm.tsx:352]: See above

File: src/components/PDF/ApplicationForm.tsx (hunk 12 above)
Comment (steffen12) [reply src/components/PDF/ApplicationForm.tsx:249]: Done

File: src/components/PDF/Applications.tsx [hunk 13]
```diff
@@ -139,43 +146,41 @@ class Applications extends Component<Props, State, {}> {
...
+  //    currentPage: 0,
+  //  });
+  // }
+  // getDocuments = () => {
+  //   const {
+  //     itemsPerPageSelected,
+  //   } = this.state;
+  //   const itemsPerPage = Number(itemsPerPageSelected.value);
+  //   // fetch call here to get all the current Documents to fill
+  // }
```
Comment (crchong1) [inline src/components/PDF/Applications.tsx:166]: do we need to keep this? or can we just delete

File: src/components/PDF/Applications.tsx [hunk 14]
```diff
@@ -139,43 +146,41 @@ class Applications extends Component<Props, State, {}> {
...
-      changeCurrentPage: this.changeCurrentPage,
-    });
+    // TODO: Pagination
+    // const itemsPerPage = Number(itemsPerPageSelected.value);
+    // const tablePageSelector = TablePageSelector({
+    //   currentPage,
+    //   itemsPerPage,
+    //   numElements,
+    //   changeCurrentPage: this.changeCurrentPage,
+    // });
```
Comment (crchong1) [inline src/components/PDF/Applications.tsx:183]: can you explain what pagination will involve? why dont we have pagination implementation in this PR?

**Review by crchong1 — APPROVED** [review]
just some small comments

File: src/components/PDF/Applications.tsx (hunk 14 above)
Comment (steffen12) [reply src/components/PDF/Applications.tsx:183]: Basically, the table pagination doesn't work

File: src/components/PDF/Applications.tsx (hunk 13 above)
Comment (steffen12) [reply src/components/PDF/Applications.tsx:166]: It might be useful for the future

File: src/components/PDF/Applications.tsx (hunk 14 above)
Comment (steffen12) [reply src/components/PDF/Applications.tsx:183]: It would need to work for the future though. Right now, there is no pagination, and it just unwraps into a long list

Comment (steffen12) [reply src/components/PDF/Applications.tsx:183]: Aren't you doing the table refactor?

Comment (loafyyy) [reply src/components/PDF/Applications.tsx:183]: Is there a table in this component? Or is the pagination just for different pages of the application? The table refactor is specifically for tables and includes the table and pagination as one component.
